File: Algorithms/1630_Arithmetic_Subarrays/solution.py
from typing import *


class Solution:
    def checkArithmeticSubarrays(self, nums: List[int], l: List[int], r: List[int]) -> List[bool]:
        return [len(set(map(lambda ys: ys[0] - ys[1], zip(xs, xs[1:])))) == 1 for xs in [sorted(nums[i:j + 1]) for i, j in zip(l, r)]]

File: Algorithms/1630_Arithmetic_Subarrays/Solution.java
import java.awt.Point;
import java.util.Arrays;
import java.util.List;
import java.util.function.BiFunction;
import java.util.stream.Collectors;
import java.util.stream.IntStream;
import java.util.stream.Stream;

class Solution {
    public List<Boolean> checkArithmeticSubarrays(int[] nums, int[] l, int[] r) {
        final BiFunction<int[], int[], Stream<Point>> zip = (a, b) -> IntStream.range(0, Math.min(a.length, b.length))
                .mapToObj(i -> new Point(a[i], b[i]));
        return zip.apply(l, r)
                .map(xs -> Arrays.copyOfRange(nums, xs.x, xs.y + 1))
                .peek(Arrays::sort)
                .map(xs -> zip.apply(xs, Arrays.copyOfRange(xs, 1, xs.length)).map(ys -> ys.x - ys.y).distinct().count() == 1)
                .collect(Collectors.toList());
    }
}